sound/soc/sof/topology.c

```diff
@@ -1275,11 +1283,23 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 	swidget->id = w->id;
 	swidget->pipeline_id = index;
 	swidget->private = NULL;
+	swidget->num_source_pins = 1;
+	swidget->num_sink_pins = 1;
+
+	ret = sof_parse_tokens(scomp, swidget, comp_pin_tokens,
+			       ARRAY_SIZE(comp_pin_tokens), priv->array,
+			       le32_to_cpu(priv->size));
+	if (ret < 0) {
+		dev_err(scomp->dev, "failed to parse component pin tokens for %s\n",
+			w->name);
+		return ret;
+	}
 
-	dev_dbg(scomp->dev, "tplg: ready widget id %d pipe %d type %d name : %s stream %s\n",
-		swidget->comp_id, index, swidget->id, tw->name,
-		strnlen(tw->sname, SNDRV_CTL_ELEM_ID_NAME_MAXLEN) > 0
-			? tw->sname : "none");
+	dev_dbg(scomp->dev,
+		"tplg: widget %d (%s) is ready [type: %d, pipe: %d, pins: %d / %d, stream: %s]\n",
+		swidget->comp_id, tw->name, swidget->id, index, swidget->num_source_pins,
```

Comment:
I don't get why you want to report tw->name here, and two lines below there's a check if tw->name is a valid string.

Reply:
I just copy and paste what was logged before, and make the debug log easier to read. `tw->name` is the widget name from topology, it is valid, and the string check two lines below is for `tw->sname`

Do you suggest to use `w->name` and `w->sname` here?